rules/linux/file_event/file_event_lnx_vshell_filename_exec.yml

```
@@ -0,0 +1,27 @@
+title: Malicious Filename with Embedded Base64 Commands - VShell
+id: 179b3686-6271-4d87-807d-17d843a8af73
+status: experimental
+description: Detects files with maliciously crafted filenames containing embedded Base64-encoded bash commands, associated with VShell malware campaigns
+references:
+    - https://www.trellix.com/blogs/research/the-silent-fileless-threat-of-vshell/
+author: "@kostastsale"
+date: 2025-08-25
+tags:
+    - attack.execution
+    - attack.t1059.004
+    - attack.defense_evasion
+    - attack.t1027
+logsource:
+    product: linux
+    category: file_event
+detection:
+    selection_filename:
+        TargetFilename|contains:
+            - '{echo'
+            - '{base64,-d}'
+    selection_generic_regex:
+        TargetFilename|re: '^.{40,}.*$'
+    condition: selection_filename or selection_generic_regex
```

```suggestion
    condition: all of selection_*
```
you probably wanted to AND the selections?

---

You didn't mention before the potential issue of the regex on matching the full path. I do see that now, you weren't clear before and I thought you were questioning the use of the regex overall.

I could change the regex to only trigger on the file name OR, as @swachchhanda000 mentioned via a comment, I can create a separate rule for that logic and remove it from this one.

---

We can definitely do that. I thought the regex should be just in addition, not it's own matching criterium as I understood it as your intention because of the PR summary: 
> The rule also includes a selection for generic regex validation to trigger only on long filenames, reducing false positives.
And I'd much more prefer it like that

---

The other idea was making a generic regex that matches on "40+ chars in between curly braces" which I would expect to be a robust detection.

---

I added a second selection for the file name because it relates to the overall goal of this detection. If everyone agrees, I can create a second rule, and as I mentioned in the comment, I can tag this PR for reference and use the same references for the report and overall description. Should I proceed with that and remove the regex filename selection from the rule in this PR?